James-Lycett/workplace-wellness
# More work on Dashboard

I got some more of the user functionality hooked up in the dashboard. You can now click profile/employees to toggle between admin view and user view (if you're an admin, of course) plus "Log new activity" and "Add a new employee" modal forms are fully functional.

To test:
- Pretty much just always 'npm i' if something isn't working when you try a new pull request, I'm gonna stop putting this in my instructions
- Log in as an admin
- Click on "Profile" then on "Employees", back and forth to see the view change
- Scope the entries table, delete an entry
- Open the "Add A New Employee" modal (only available on admin view, i.e. after clicking 'employees')
- Add a new employee, make sure it's in the employees table
- Open "Log New Activity" modal
- Log a new activity, make sure it's in the entries table

Changes:
- Combined Dashboard loadData() api calls into one single call
- Added api call stuff to Modal onSubmit functions
- Made some minor modifications to the Dashboard conditional rendering, mostly just cleaner code
- Some of the sidebar links are still empty, I didn't know what to do with them. 
- Plugged some averages into a couple of the progress charts at the top of the user view.
- Modified 'ActivityLogModal' to just 'Modal', used for both activity log and add new employee depending on what clicked you there.
- Lifted up the 'employees' state from EmployeesListNew (and everything that goes with it) to Dashboard so it won't re-fetch every time we toggle between views
- Some final touches on the functionality of the entries table - delete button works now, 'edit' column is removed, 'date' column actually loads formatted dates
- Fixed some backend validation stuff and added a little more error handling back there
